### ISSUE TITLE
Use Keystone for assembling?

### ISSUE THREAD

**Issue by aquynh** [yrp604/rappel#6]
what do you think about using Keystone for assembling for Rappel, which offers more flexibility as a true framework?

http://www.keystone-engine.org/

cheers.

**Comment by yrp604** [comment]
Yep, this is a good idea and I'd like to implement it eventually. I'm unlikely to do it in the near future just due to time constraints, but it's definitely on my radar, especially as I expand architecture support.

**Comment by aquynh** [comment]
Awesome, keep me updated!

**Comment by aquynh** [comment]
Also, it will be nice to see Rappel supports Unicorn emulator as a backend,
too  :-)

**Comment by yrp604** [comment]
I'll definitely keep you updated.

If I add Unicorn support I'll likely create a new project for it and repurpose some rappel UI and ELF gen code, then keep the two separate. I'd rather not conflate the two, especially because people using hardware for semantic recovery of instructions to find emulator bugs one of the main use cases of rappel currently.

**Comment by shiftre** [comment]
Simple question, why not use the unicorn infrastrucutre instead of nasm and let keystone handle everythhing?

**Comment by shiftre** [comment]
I'd like to contribute this sort of integration, how would keystone be integrated to replace the nasm infrastructure?

**Comment by omega13-engr** [comment]
I made it using python: https://github.com/slaee/pyrappel
